{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.11.5 (main, Sep 11 2023, 13:54:46) [GCC 11.2.0]\n"
     ]
    }
   ],
   "source": [
    "import sys\n",
    "print(sys.version)\n",
    "sys.path.append('/Users/geba9152/LIET/liet/')\n",
    "\n",
    "from liet_res_class import FitParse, fitparse_intersect\n",
    "\n",
    "import pandas as pd\n",
    "import statsmodels.api as sm\n",
    "from statsmodels.stats.anova import AnovaRM\n",
    "from statsmodels.stats.multicomp import pairwise_tukeyhsd\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Date: 06/12/2024\n",
    "\n",
    "This code is adapted from Jacob's code\n",
    "\n",
    "I fixed the problematic PAS vs vs 5' UTR/last exon reference problem.\n",
    "\n",
    "Here I am running an ANOVA in all perturbed and contorl meta samples to detect global changes in posterior values"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ANOVA data formatter function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def anova_data_formatter(param, fit_parses, treatments):\n",
    "#     assert len(fit_parses) == len(treatments), \"Number of treatments must match number of FitParse objects.\"\n",
    "#     assert all(len(i.genes) == len(fit_parses[0].genes) for i in fit_parses), \"FitParse objects have different genes.\"\n",
    "    \n",
    "    gene_ids = fit_parses[0].genes\n",
    "    \n",
    "    subject_array = [g for g in gene_ids for _ in range(len(treatments))]\n",
    "    treatment_array = treatments * len(gene_ids)\n",
    "    \n",
    "    response_array = [fitp.__dict__[param] for fitp in fit_parses]\n",
    "    response_array = [i for fitp_list in zip(*response_array) for i in fitp_list]\n",
    "    \n",
    "    # Check if the lengths of subject_array, treatment_array, and response_array are not equal\n",
    "    if len(subject_array) != len(response_array) or len(treatment_array) != len(response_array):\n",
    "        # Create a list of valid indices where the index is less than the length of response_array\n",
    "        valid_indices = [i for i in range(len(subject_array)) if i < len(response_array)]\n",
    "        \n",
    "        # Filter subject_array to include only the elements at the valid indices\n",
    "        subject_array = [subject_array[i] for i in valid_indices]\n",
    "        \n",
    "        # Filter treatment_array to include only the elements at the valid indices\n",
    "        treatment_array = [treatment_array[i] for i in valid_indices]\n",
    "        \n",
    "        # Filter response_array to include only the elements at the valid indices\n",
    "        response_array = [response_array[i] for i in valid_indices]\n",
    "\n",
    "    \n",
    "    # Debug statements to check lengths\n",
    "    print(f'Length of subject_array (genes): {len(subject_array)}')\n",
    "    print(f'Length of treatment_array: {len(treatment_array)}')\n",
    "    print(f'Length of response_array (posterior val): {len(response_array)}')\n",
    "\n",
    "    data = {\n",
    "        'subject': subject_array,\n",
    "        'treatment': treatment_array,\n",
    "        param: response_array\n",
    "    }\n",
    "    \n",
    "    df = pd.DataFrame(data)\n",
    "    \n",
    "    if param == 'mT':\n",
    "        ann = pd.read_csv(\"/scratch/Users/geba9152/LIET-summer2024/meta-celltype-comparison/annotation/lietanns/chr1-6-3p-UTR.liet.ann\", header=None, sep=\"\\t\")\n",
    "        ann.columns = ['chr', 'start', 'PAS', 'subject', 'length', 'strand']\n",
    "        ann = ann[['subject', 'length']]\n",
    "        df_merged = df.merge(ann, on='subject')\n",
    "        df_merged['mT'] = df_merged['mT'] - df_merged['length']\n",
    "        df = df_merged[['subject', 'treatment', 'mT']]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Ethan/Eric heat shock data\n",
    "- 42 degrees C, 1 hour "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "liet_res_dir = '/scratch/Shares/dowell/for_jacob/from_georgia/ethan-eric-hs-results-05142024/results/'\n",
    "\n",
    "# er/et (Eric/Ethan), 1/2 (rep1/rep2), c/h (control/heat shock)\n",
    "er1c = liet_res_dir+'Eric37_rep1.liet'\n",
    "er2c = liet_res_dir+'Eric37_rep2.liet'\n",
    "et1c = liet_res_dir+'Ethan37_rep1.liet'\n",
    "et2c = liet_res_dir+'Ethan37_rep2.liet'\n",
    "er1h = liet_res_dir+'Eric42_rep1.liet'\n",
    "er2h = liet_res_dir+'Eric42_rep2.liet'\n",
    "et1h = liet_res_dir+'Ethan42_rep1.liet'\n",
    "et2h = liet_res_dir+'Ethan42_rep2.liet'\n",
    "\n",
    "# d/t disomy (Eric) or Trisomic (Ethan)\n",
    "d1c = FitParse(er1c, log_file=er1c+'.log')\n",
    "d2c = FitParse(er2c, log_file=er2c+'.log')\n",
    "t1c = FitParse(et1c, log_file=et1c+'.log')\n",
    "t2c = FitParse(et2c, log_file=et2c+'.log')\n",
    "d1h = FitParse(er1h, log_file=er1h+'.log')\n",
    "d2h = FitParse(er2h, log_file=er2h+'.log')\n",
    "t1h = FitParse(et1h, log_file=et1h+'.log')\n",
    "t2h = FitParse(et2h, log_file=et2h+'.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of subject_array (genes): 1304\n",
      "Length of treatment_array: 1304\n",
      "Length of response_array (posterior val): 1304\n",
      "                 Anova\n",
      "========================================\n",
      "          F Value Num DF  Den DF  Pr > F\n",
      "----------------------------------------\n",
      "treatment 42.2969 3.0000 486.0000 0.0000\n",
      "========================================\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:509: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  .groupby([self.subject] + self.within,\n",
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:511: FutureWarning: The provided callable <function mean at 0x2ae0d3c4fd80> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string \"mean\" instead.\n",
      "  .agg(self.aggregate_func))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Multiple Comparison of Means - Tukey HSD, FWER=0.05    \n",
      "===========================================================\n",
      "group1 group2  meandiff  p-adj    lower      upper   reject\n",
      "-----------------------------------------------------------\n",
      "     D    D+H  1996.4229 0.0001   788.6065 3204.2392   True\n",
      "     D      T   -92.6669 0.9973 -1300.4832 1115.1495  False\n",
      "     D    T+H  1988.6583 0.0001   780.8419 3196.4746   True\n",
      "   D+H      T -2089.0897 0.0001 -3296.9061 -881.2734   True\n",
      "   D+H    T+H    -7.7646    1.0 -1215.5809 1200.0518  False\n",
      "     T    T+H  2081.3252 0.0001   873.5088 3289.1415   True\n",
      "-----------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "liet_param = 'mT'\n",
    "fit_parses = [d1c, d2c, t1c, t2c, d1h, d2h, t1h, t2h]\n",
    "treatments = ['D', 'D', 'T', 'T', 'D+H', 'D+H', 'T+H', 'T+H']\n",
    "\n",
    "anova_inputs = anova_data_formatter(liet_param, fit_parses, treatments)\n",
    "anova_inputs['subject'] = anova_inputs['subject'].astype('category')\n",
    "\n",
    "aovrm = AnovaRM(anova_inputs, liet_param, 'subject', within=['treatment'], aggregate_func='mean')\n",
    "\n",
    "res = aovrm.fit()\n",
    "print(res)\n",
    "\n",
    "# Perform Tukey's HSD post-hoc test\n",
    "tukey = pairwise_tukeyhsd(endog=anova_inputs[liet_param], groups=anova_inputs['treatment'], alpha=0.05)\n",
    "\n",
    "# Display the results\n",
    "print(tukey)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CDK7i + Heat Shock OV90\n",
    "\n",
    "- All drug treated data: 30 min pre-treatment prior to any heat shock (42 degrees C)\n",
    "- Concentration of SY5609 = 50 nM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['assay', 'sample_read1', 'sample_read2', 'id', 'idx', 'coded_reps',\n",
       "       'replicate', 'batch', 'advanced_batch', 'genotype', 'drug_pretreatment',\n",
       "       'hs_time', 'recovery_time', 'total_time', 'tx_type', 'group',\n",
       "       'millionsmapped', 'spikein', 'spikein_mm', 'VSI', 'VSI_v1', 'linear_sf',\n",
       "       'deseq_sf', 'dm6_VS', 'dm6_VS_v1', 'dm6_linear_sf', 'dm6_deseq2_sf',\n",
       "       'ercc_ctrlgenes_deseq2_sf', 'fastq_path', 'analysis_path', 'path_sep',\n",
       "       'color', 'millionsmapped_updated', 'deseq2_ercc_norm'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = pd.read_csv('/scratch/Shares/dowell/ov90_cdk7i/PROseq/ov90_cdk7i_metadata_PRO.txt', sep = \"\\t\")\n",
    "meta.columns\n",
    "# meta['drug_pretreatment']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "liet_res_dir = '/scratch/Users/geba9152/LIET-summer2024/cdk7i-ov90/results/tight-sT/'\n",
    "\n",
    "# liet paths\n",
    "dmso_0_1_path = liet_res_dir+'DMSO_0_R1.liet'\n",
    "dmso_0_2_path = liet_res_dir+'DMSO_0_R2.liet'\n",
    "dmso_20_1_path = liet_res_dir+'DMSO_20_R1.liet'\n",
    "dmso_20_2_path = liet_res_dir+'DMSO_20_R2.liet'\n",
    "dmso_45_1_path = liet_res_dir+'DMSO_45_R1.liet'\n",
    "dmso_45_2_path = liet_res_dir+'DMSO_45_R2.liet'\n",
    "\n",
    "sy_0_1_path = liet_res_dir+'SY_0_R1.liet'\n",
    "sy_0_2_path = liet_res_dir+'SY_0_R2.liet'\n",
    "sy_20_1_path = liet_res_dir+'SY_20_R1.liet'\n",
    "sy_20_2_path = liet_res_dir+'SY_20_R2.liet'\n",
    "sy_45_1_path = liet_res_dir+'SY_45_R1.liet'\n",
    "sy_45_2_path = liet_res_dir+'SY_45_R2.liet'\n",
    "\n",
    "\n",
    "# fp objects\n",
    "dmso_0_1 = FitParse(dmso_0_1_path, log_file=dmso_0_1_path+'.log')\n",
    "dmso_0_2 = FitParse(dmso_0_2_path, log_file=dmso_0_2_path+'.log')\n",
    "dmso_20_1 = FitParse(dmso_20_1_path, log_file=dmso_20_1_path+'.log')\n",
    "dmso_20_2 = FitParse(dmso_20_2_path, log_file=dmso_20_2_path+'.log')\n",
    "dmso_45_1 = FitParse(dmso_45_1_path, log_file=dmso_45_1_path+'.log')\n",
    "dmso_45_2 = FitParse(dmso_45_2_path, log_file=dmso_45_2_path+'.log')\n",
    "\n",
    "sy_0_1 = FitParse(sy_0_1_path, log_file=sy_0_1_path+'.log')\n",
    "sy_0_2 = FitParse(sy_0_2_path, log_file=sy_0_2_path+'.log')\n",
    "sy_20_1 = FitParse(sy_20_1_path, log_file=sy_20_1_path+'.log')\n",
    "sy_20_2 = FitParse(sy_20_2_path, log_file=sy_20_2_path+'.log')\n",
    "sy_45_1 = FitParse(sy_45_1_path, log_file=sy_45_1_path+'.log')\n",
    "sy_45_2 = FitParse(sy_45_2_path, log_file=sy_45_2_path+'.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of subject_array (genes): 1956\n",
      "Length of treatment_array: 1956\n",
      "Length of response_array (posterior val): 1956\n",
      "                 Anova\n",
      "========================================\n",
      "          F Value Num DF  Den DF  Pr > F\n",
      "----------------------------------------\n",
      "treatment 15.8938 5.0000 810.0000 0.0000\n",
      "========================================\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:509: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  .groupby([self.subject] + self.within,\n",
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:511: FutureWarning: The provided callable <function mean at 0x2ae0d3c4fd80> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string \"mean\" instead.\n",
      "  .agg(self.aggregate_func))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Multiple Comparison of Means - Tukey HSD, FWER=0.05     \n",
      "============================================================\n",
      " group1  group2  meandiff p-adj    lower      upper   reject\n",
      "------------------------------------------------------------\n",
      " DMSO_0 DMSO_20  825.2788 0.0002   283.2613 1367.2963   True\n",
      " DMSO_0 DMSO_45 1100.7294    0.0    558.712 1642.7469   True\n",
      " DMSO_0    SY_0  122.3052 0.9877  -419.7123  664.3227  False\n",
      " DMSO_0   SY_20   910.751    0.0   368.7336 1452.7685   True\n",
      " DMSO_0   SY_45  819.9923 0.0002   277.9748 1362.0098   True\n",
      "DMSO_20 DMSO_45  275.4506 0.6964  -266.5668  817.4681  False\n",
      "DMSO_20    SY_0 -702.9736  0.003 -1244.9911 -160.9562   True\n",
      "DMSO_20   SY_20   85.4722 0.9977  -456.5452  627.4897  False\n",
      "DMSO_20   SY_45   -5.2865    1.0   -547.304   536.731  False\n",
      "DMSO_45    SY_0 -978.4243    0.0 -1520.4417 -436.4068   True\n",
      "DMSO_45   SY_20 -189.9784 0.9181  -731.9959  352.0391  False\n",
      "DMSO_45   SY_45 -280.7371 0.6788  -822.7546  261.2803  False\n",
      "   SY_0   SY_20  788.4459 0.0005   246.4284 1330.4633   True\n",
      "   SY_0   SY_45  697.6871 0.0034   155.6696 1239.7046   True\n",
      "  SY_20   SY_45  -90.7587 0.9969  -632.7762  451.2587  False\n",
      "------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "liet_param = 'sT'\n",
    "\n",
    "fit_parses = [dmso_0_1, dmso_0_2, dmso_20_1, dmso_20_2, dmso_45_1, dmso_45_2, sy_0_1, sy_0_2, sy_20_1, sy_20_2, sy_45_1, sy_45_2]\n",
    "treatments = ['DMSO_0', 'DMSO_0','DMSO_20', 'DMSO_20','DMSO_45','DMSO_45','SY_0', 'SY_0','SY_20', 'SY_20','SY_45','SY_45']\n",
    "\n",
    "anova_inputs = anova_data_formatter(liet_param, fit_parses, treatments)\n",
    "anova_inputs['subject'] = anova_inputs['subject'].astype('category')\n",
    "\n",
    "aovrm = AnovaRM(anova_inputs, liet_param, 'subject', within=['treatment'], aggregate_func='mean')\n",
    "\n",
    "res = aovrm.fit()\n",
    "print(res)\n",
    "\n",
    "# Perform Tukey's HSD post-hoc test\n",
    "tukey = pairwise_tukeyhsd(endog=anova_inputs[liet_param], groups=anova_inputs['treatment'], alpha=0.05)\n",
    "\n",
    "# Display the results\n",
    "print(tukey)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# CDK7i SY5609 HCT\n",
    "\n",
    "`#TODO: run LIET on the rest of this data`\n",
    "\n",
    "- 50nM SY5609\n",
    "- 100nM KB0742\n",
    "- IFN gamma was used at 10ng/mL\n",
    "\n",
    "Combo scheme is as follows: time 0min—cells are treated with DMSO or inhibitor/s, time 30min—IFN gamma is added, time 75min—cells are harvested."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "liet_res_dir = '/scratch/Users/geba9152/LIET-summer2024/cdk7i-hct/results/tight-sT/'\n",
    "\n",
    "# liet paths\n",
    "c1_path = liet_res_dir+'DMSO_1.liet'\n",
    "c2_path = liet_res_dir+'DMSO_2.liet'\n",
    "sy5609_1_path = liet_res_dir+'SY5609_1.liet'\n",
    "sy5609_2_path = liet_res_dir+'SY5609_2.liet'\n",
    "\n",
    "# fp objects\n",
    "c1 = FitParse(c1_path, log_file=c1_path+'.log')\n",
    "c2 = FitParse(c2_path, log_file=c2_path+'.log')\n",
    "s1 = FitParse(sy5609_1_path, log_file=sy5609_1_path+'.log')\n",
    "s2 = FitParse(sy5609_2_path, log_file=sy5609_2_path+'.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of subject_array (genes): 652\n",
      "Length of treatment_array: 652\n",
      "Length of response_array (posterior val): 652\n",
      "                 Anova\n",
      "========================================\n",
      "          F Value Num DF  Den DF  Pr > F\n",
      "----------------------------------------\n",
      "treatment 60.1285 1.0000 162.0000 0.0000\n",
      "========================================\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:509: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  .groupby([self.subject] + self.within,\n",
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:511: FutureWarning: The provided callable <function mean at 0x2ae0d3c4fd80> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string \"mean\" instead.\n",
      "  .agg(self.aggregate_func))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     Multiple Comparison of Means - Tukey HSD, FWER=0.05      \n",
      "==============================================================\n",
      "group1    group2     meandiff p-adj   lower     upper   reject\n",
      "--------------------------------------------------------------\n",
      "     C CDK7i-noncov 2842.8713   0.0 1478.5224 4207.2202   True\n",
      "--------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "liet_param = 'mT'\n",
    "\n",
    "fit_parses = [c1, c2, s1, s2]\n",
    "treatments = ['C', 'C','CDK7i-noncov', 'CDK7i-noncov']\n",
    "\n",
    "anova_inputs = anova_data_formatter(liet_param, fit_parses, treatments)\n",
    "anova_inputs['subject'] = anova_inputs['subject'].astype('category')\n",
    "\n",
    "aovrm = AnovaRM(anova_inputs, liet_param, 'subject', within=['treatment'], aggregate_func='mean')\n",
    "\n",
    "res = aovrm.fit()\n",
    "print(res)\n",
    "\n",
    "# Perform Tukey's HSD post-hoc test\n",
    "tukey = pairwise_tukeyhsd(endog=anova_inputs[liet_param], groups=anova_inputs['treatment'], alpha=0.05)\n",
    "\n",
    "# Display the results\n",
    "print(tukey)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# IFN/CDK7i (SY351)/CDK9i HCT\n",
    "- 50nM SY351\n",
    "- 100nM KB0742 \n",
    "- IFN gamma was used at 10ng/mL\n",
    "\n",
    "Combo scheme is as follows: time 0min—cells are treated with DMSO or inhibitor/s, time 30min—IFN gamma is added, time 75min—cells are harvested."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "liet_res_dir = '/scratch/Users/geba9152/LIET-summer2024/KB0742-SY351-batch/results/tight-sT/'\n",
    "\n",
    "# liet paths\n",
    "c1_path = liet_res_dir+'DMSO_1_S1_L001_R1_001.liet'\n",
    "c2_path = liet_res_dir+'DMSO_2_S6_L001_R1_001.liet'\n",
    "ifn1_path = liet_res_dir+'IFN_1.liet'\n",
    "ifn2_path = liet_res_dir+'IFN_2.liet'\n",
    "cdk91_path = liet_res_dir+'KB0742_1_S4_L001_R1_001.liet'\n",
    "cdk92_path = liet_res_dir+'KB0742_2_S9_L001_R1_001.liet'\n",
    "cdk71_path = liet_res_dir+'SY351_1_S3_L001_R1_001.liet'\n",
    "cdk72_path = liet_res_dir+'SY351_2_S8_L001_R1_001.liet'\n",
    "cdk971_path = liet_res_dir+'SY-KB_1_S1_L001_R1_001.liet'\n",
    "cdk972_path = liet_res_dir+'SY-KB_1_S5_L001_R1_001.liet'\n",
    "cdk973_path = liet_res_dir+'SY-KB_2_S10_L001_R1_001.liet'\n",
    "\n",
    "# fp objects\n",
    "c1 = FitParse(c1_path, log_file=c1_path+'.log')\n",
    "c2 = FitParse(c2_path, log_file=c2_path+'.log')\n",
    "ifn1 = FitParse(ifn1_path, log_file=ifn1_path+'.log')\n",
    "ifn2 = FitParse(ifn2_path, log_file=ifn2_path+'.log')\n",
    "cdk91 = FitParse(cdk91_path, log_file=cdk91_path+'.log')\n",
    "cdk92 = FitParse(cdk92_path, log_file=cdk92_path+'.log')\n",
    "cdk71 = FitParse(cdk71_path, log_file=cdk71_path+'.log')\n",
    "cdk72 = FitParse(cdk72_path, log_file=cdk72_path+'.log')\n",
    "cdk971 = FitParse(cdk971_path, log_file=cdk971_path+'.log')\n",
    "cdk972 = FitParse(cdk972_path, log_file=cdk972_path+'.log')\n",
    "cdk973 = FitParse(cdk973_path, log_file=cdk973_path+'.log')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of subject_array (genes): 1749\n",
      "Length of treatment_array: 1749\n",
      "Length of response_array (posterior val): 1749\n",
      "                 Anova\n",
      "========================================\n",
      "          F Value Num DF  Den DF  Pr > F\n",
      "----------------------------------------\n",
      "treatment  0.2867 4.0000 632.0000 0.8866\n",
      "========================================\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:509: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.\n",
      "  .groupby([self.subject] + self.within,\n",
      "/Users/geba9152/miniconda3/envs/statsmodels/lib/python3.11/site-packages/statsmodels/stats/anova.py:511: FutureWarning: The provided callable <function mean at 0x2ae0d3c4fd80> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string \"mean\" instead.\n",
      "  .agg(self.aggregate_func))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Multiple Comparison of Means - Tukey HSD, FWER=0.05    \n",
      "===========================================================\n",
      "group1 group2  meandiff  p-adj    lower      upper   reject\n",
      "-----------------------------------------------------------\n",
      "     C   CDK7  -135.5285    1.0 -8340.0746 8069.0176  False\n",
      "     C CDK7+9 -2241.6192 0.9254 -9731.3108 5248.0725  False\n",
      "     C   CDK9 -1125.4688 0.9958  -9330.015 7079.0773  False\n",
      "     C    IFN  -371.4239 0.9999   -8575.97 7833.1223  False\n",
      "  CDK7 CDK7+9 -2106.0907 0.9398 -9595.7823  5383.601  False\n",
      "  CDK7   CDK9  -989.9403 0.9975 -9194.4865 7214.6058  False\n",
      "  CDK7    IFN  -235.8954    1.0 -8440.4415 7968.6508  False\n",
      "CDK7+9   CDK9  1116.1503 0.9942 -6373.5413  8605.842  False\n",
      "CDK7+9    IFN  1870.1953 0.9604 -5619.4964 9359.8869  False\n",
      "  CDK9    IFN    754.045 0.9991 -7450.5012 8958.5911  False\n",
      "-----------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "liet_param = 'mT'\n",
    "fit_parses = [c1, c2, ifn1, ifn2, cdk91, cdk92, cdk71, cdk72, cdk971, cdk972, cdk973]\n",
    "treatments = ['C', 'C', 'IFN', 'IFN', 'CDK9', 'CDK9', 'CDK7', 'CDK7', 'CDK7+9', 'CDK7+9', 'CDK7+9']\n",
    "\n",
    "# fit_parses = [c1, c2, ifn1, ifn2, cdk91, cdk92, cdk71, cdk72, cdk971, cdk972]\n",
    "# treatments = ['C', 'C', 'IFN', 'IFN', 'CDK9', 'CDK9', 'CDK7', 'CDK7', 'CDK7+9', 'CDK7+9']\n",
    "\n",
    "\n",
    "anova_inputs = anova_data_formatter(liet_param, fit_parses, treatments)\n",
    "anova_inputs['subject'] = anova_inputs['subject'].astype('category')\n",
    "\n",
    "aovrm = AnovaRM(anova_inputs, liet_param, 'subject', within=['treatment'], aggregate_func='mean')\n",
    "\n",
    "res = aovrm.fit()\n",
    "print(res)\n",
    "\n",
    "# Perform Tukey's HSD post-hoc test\n",
    "tukey = pairwise_tukeyhsd(endog=anova_inputs[liet_param], groups=anova_inputs['treatment'], alpha=0.05)\n",
    "\n",
    "# Display the results\n",
    "print(tukey)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Meta cross cell type comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "CHECK LINE: ['NUCKS1|NM_022731.5: fitting error']\n",
      "CHECK LINE: ['MAN1A2|NM_006699.5: fitting error']\n",
      "CHECK LINE: ['MAN1A2|NM_006699.5: fitting error']\n",
      "CHECK LINE: ['NUCKS1|NM_022731.5: fitting error']\n",
      "CHECK LINE: ['RPL37A|NM_000998.5: fitting error']\n"
     ]
    }
   ],
   "source": [
    "liet_res_dir = '/scratch/Users/geba9152/LIET-summer2024/meta-celltype-comparison/results/tight-sT/'\n",
    "\n",
    "# liet paths\n",
    "meta_OCILY1 = liet_res_dir+'meta_OCILY1.liet'\n",
    "meta_G401 = liet_res_dir+'meta_G401.liet'\n",
    "meta_HCT116 = liet_res_dir+'meta_HCT116.liet'\n",
    "meta_MEL624 = liet_res_dir+'meta_MEL624.liet'\n",
    "meta_HeLa = liet_res_dir+'meta_HeLa.liet'\n",
    "meta_THP1 = liet_res_dir+'meta_THP1.liet'\n",
    "meta_foreskin_fibroblast = liet_res_dir+'meta_foreskin_fibroblast.liet'\n",
    "meta_A375 = liet_res_dir+'meta_A375.liet'\n",
    "meta_SUDHL4 = liet_res_dir+'meta_SUDHL4.liet'\n",
    "meta_lymphoblast = liet_res_dir+'meta_lymphoblast.liet'\n",
    "meta_Jurkat_T_cell = liet_res_dir+'meta_Jurkat_T_cell.liet'\n",
    "meta_ESC = liet_res_dir+'meta_ESC.liet'\n",
    "meta_LC2ad = liet_res_dir+'meta_LC2ad.liet'\n",
    "meta_Ramos = liet_res_dir+'meta_Ramos.liet'\n",
    "meta_HEK239T_HEK239 = liet_res_dir+'meta_HEK239T-HEK239.liet'\n",
    "meta_BEAS2B = liet_res_dir+'meta_BEAS2B.liet'\n",
    "meta_NUDUL1 = liet_res_dir+'meta_NUDUL1.liet'\n",
    "meta_K562 = liet_res_dir+'meta_K562.liet'\n",
    "meta_MCF7 = liet_res_dir+'meta_MCF7.liet'\n",
    "meta_Kasumi1 = liet_res_dir+'meta_Kasumi1.liet'\n",
    "meta_S2VP10 = liet_res_dir+'meta_S2VP10.liet'\n",
    "meta_KBM7 = liet_res_dir+'meta_KBM7.liet'\n",
    "meta_CD34_erythroblast = liet_res_dir+'meta_CD34_erythroblast.liet'\n",
    "meta_HAP1 = liet_res_dir+'meta_HAP1.liet'\n",
    "meta_U936 = liet_res_dir+'meta_U936.liet'\n",
    "meta_CD4_T_cell = liet_res_dir+'meta_CD4_T_cell.liet'\n",
    "meta_A549 = liet_res_dir+'meta_A549.liet'\n",
    "\n",
    "# fp objects\n",
    "meta_OCILY1_fp = FitParse(meta_OCILY1, log_file=meta_OCILY1+'.log')\n",
    "meta_G401_fp = FitParse(meta_G401, log_file=meta_G401+'.log')\n",
    "meta_HCT116_fp = FitParse(meta_HCT116, log_file=meta_HCT116+'.log')\n",
    "meta_MEL624_fp = FitParse(meta_MEL624, log_file=meta_MEL624+'.log')\n",
    "meta_HeLa_fp = FitParse(meta_HeLa, log_file=meta_HeLa+'.log')\n",
    "meta_THP1_fp = FitParse(meta_THP1, log_file=meta_THP1+'.log')\n",
    "meta_foreskin_fibroblast_fp = FitParse(meta_foreskin_fibroblast, log_file=meta_foreskin_fibroblast+'.log')\n",
    "meta_A375_fp = FitParse(meta_A375, log_file=meta_A375+'.log')\n",
    "meta_SUDHL4_fp = FitParse(meta_SUDHL4, log_file=meta_SUDHL4+'.log')\n",
    "meta_lymphoblast_fp = FitParse(meta_lymphoblast, log_file=meta_lymphoblast+'.log')\n",
    "meta_Jurkat_T_cell_fp = FitParse(meta_Jurkat_T_cell, log_file=meta_Jurkat_T_cell+'.log')\n",
    "meta_ESC_fp = FitParse(meta_ESC, log_file=meta_ESC+'.log')\n",
    "meta_LC2ad_fp = FitParse(meta_LC2ad, log_file=meta_LC2ad+'.log')\n",
    "meta_Ramos_fp = FitParse(meta_Ramos, log_file=meta_Ramos+'.log')\n",
    "meta_HEK239T_HEK239_fp = FitParse(meta_HEK239T_HEK239, log_file=meta_HEK239T_HEK239+'.log')\n",
    "meta_BEAS2B_fp = FitParse(meta_BEAS2B, log_file=meta_BEAS2B+'.log')\n",
    "meta_NUDUL1_fp = FitParse(meta_NUDUL1, log_file=meta_NUDUL1+'.log')\n",
    "meta_K562_fp = FitParse(meta_K562, log_file=meta_K562+'.log')\n",
    "meta_MCF7_fp = FitParse(meta_MCF7, log_file=meta_MCF7+'.log')\n",
    "meta_Kasumi1_fp = FitParse(meta_Kasumi1, log_file=meta_Kasumi1+'.log')\n",
    "meta_S2VP10_fp = FitParse(meta_S2VP10, log_file=meta_S2VP10+'.log')\n",
    "meta_KBM7_fp = FitParse(meta_KBM7, log_file=meta_KBM7+'.log')\n",
    "meta_CD34_erythroblast_fp = FitParse(meta_CD34_erythroblast, log_file=meta_CD34_erythroblast+'.log')\n",
    "meta_HAP1_fp = FitParse(meta_HAP1, log_file=meta_HAP1+'.log')\n",
    "meta_U936_fp = FitParse(meta_U936, log_file=meta_U936+'.log')\n",
    "meta_CD4_T_cell_fp = FitParse(meta_CD4_T_cell, log_file=meta_CD4_T_cell+'.log')\n",
    "meta_A549_fp = FitParse(meta_A549, log_file=meta_A549+'.log')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27\n",
      "Length of subject_array (genes): 4347\n",
      "Length of treatment_array: 4347\n",
      "Length of response_array (posterior val): 4347\n",
      "                  Anova\n",
      "==========================================\n",
      "          F Value  Num DF   Den DF  Pr > F\n",
      "------------------------------------------\n",
      "treatment  4.0602 26.0000 4160.0000 0.0000\n",
      "==========================================\n",
      "\n",
      "                      Multiple Comparison of Means - Tukey HSD, FWER=0.05                      \n",
      "===============================================================================================\n",
      "         group1                   group2           meandiff  p-adj    lower      upper   reject\n",
      "-----------------------------------------------------------------------------------------------\n",
      "               meta_A375                meta_A549  -345.8922    1.0 -1578.4755   886.691  False\n",
      "               meta_A375              meta_BEAS2B  -330.4082    1.0 -1562.9914   902.175  False\n",
      "               meta_A375   meta_CD34_erythroblast  -300.6593    1.0 -1533.2425  931.9239  False\n",
      "               meta_A375          meta_CD4_T_cell  -336.5102    1.0 -1569.0934   896.073  False\n",
      "               meta_A375                 meta_ESC  -200.5291    1.0 -1433.1123 1032.0541  False\n",
      "               meta_A375                meta_G401  -445.3845 0.9999 -1677.9678  787.1987  False\n",
      "               meta_A375                meta_HAP1  -165.9402    1.0 -1398.5234  1066.643  False\n",
      "               meta_A375              meta_HCT116  -305.2676    1.0 -1537.8509  927.3156  False\n",
      "               meta_A375      meta_HEK239T_HEK239   102.7199    1.0 -1129.8633 1335.3031  False\n",
      "               meta_A375                meta_HeLa  -390.2052    1.0 -1622.7884   842.378  False\n",
      "               meta_A375       meta_Jurkat_T_cell   732.8373 0.9074  -499.7459 1965.4205  False\n",
      "               meta_A375                meta_K562  -529.7646 0.9986 -1762.3478  702.8186  False\n",
      "               meta_A375                meta_KBM7  -312.9739    1.0 -1545.5571  919.6093  False\n",
      "               meta_A375             meta_Kasumi1  -531.0379 0.9986 -1763.6211  701.5453  False\n",
      "               meta_A375               meta_LC2ad  -284.3264    1.0 -1516.9096  948.2568  False\n",
      "               meta_A375                meta_MCF7  -312.8635    1.0 -1545.4467  919.7197  False\n",
      "               meta_A375              meta_MEL624  -296.3291    1.0 -1528.9123  936.2541  False\n",
      "               meta_A375              meta_NUDUL1  -218.8811    1.0 -1451.4643 1013.7021  False\n",
      "               meta_A375              meta_OCILY1   543.9493 0.9979   -688.634 1776.5325  False\n",
      "               meta_A375               meta_Ramos  -528.9443 0.9987 -1761.5276  703.6389  False\n",
      "               meta_A375              meta_S2VP10  -324.9434    1.0 -1557.5266  907.6398  False\n",
      "               meta_A375              meta_SUDHL4  -764.3135 0.8636 -1996.8968  468.2697  False\n",
      "               meta_A375                meta_THP1  -685.9306 0.9534 -1918.5138  546.6526  False\n",
      "               meta_A375                meta_U936 -1033.9435 0.2721 -2266.5267  198.6397  False\n",
      "               meta_A375 meta_foreskin_fibroblast   -82.4485    1.0 -1315.0317 1150.1347  False\n",
      "               meta_A375         meta_lymphoblast  -633.5299 0.9819 -1866.1131  599.0533  False\n",
      "               meta_A549              meta_BEAS2B     15.484    1.0 -1217.0992 1248.0673  False\n",
      "               meta_A549   meta_CD34_erythroblast    45.2329    1.0 -1187.3503 1277.8161  False\n",
      "               meta_A549          meta_CD4_T_cell      9.382    1.0 -1223.2012 1241.9653  False\n",
      "               meta_A549                 meta_ESC   145.3631    1.0 -1087.2201 1377.9463  False\n",
      "               meta_A549                meta_G401   -99.4923    1.0 -1332.0755 1133.0909  False\n",
      "               meta_A549                meta_HAP1    179.952    1.0 -1052.6312 1412.5353  False\n",
      "               meta_A549              meta_HCT116    40.6246    1.0 -1191.9586 1273.2078  False\n",
      "               meta_A549      meta_HEK239T_HEK239   448.6121 0.9999  -783.9711 1681.1953  False\n",
      "               meta_A549                meta_HeLa    -44.313    1.0 -1276.8962 1188.2702  False\n",
      "               meta_A549       meta_Jurkat_T_cell  1078.7296 0.1966  -153.8537 2311.3128  False\n",
      "               meta_A549                meta_K562  -183.8724    1.0 -1416.4556 1048.7109  False\n",
      "               meta_A549                meta_KBM7    32.9183    1.0 -1199.6649 1265.5015  False\n",
      "               meta_A549             meta_Kasumi1  -185.1457    1.0 -1417.7289 1047.4376  False\n",
      "               meta_A549               meta_LC2ad    61.5658    1.0 -1171.0174 1294.1491  False\n",
      "               meta_A549                meta_MCF7    33.0288    1.0 -1199.5545  1265.612  False\n",
      "               meta_A549              meta_MEL624    49.5631    1.0 -1183.0201 1282.1463  False\n",
      "               meta_A549              meta_NUDUL1   127.0111    1.0 -1105.5721 1359.5943  False\n",
      "               meta_A549              meta_OCILY1   889.8415 0.5997  -342.7417 2122.4247  False\n",
      "               meta_A549               meta_Ramos  -183.0521    1.0 -1415.6353 1049.5311  False\n",
      "               meta_A549              meta_S2VP10    20.9488    1.0 -1211.6344  1253.532  False\n",
      "               meta_A549              meta_SUDHL4  -418.4213    1.0 -1651.0045  814.1619  False\n",
      "               meta_A549                meta_THP1  -340.0384    1.0 -1572.6216  892.5448  False\n",
      "               meta_A549                meta_U936  -688.0512 0.9518 -1920.6345   544.532  False\n",
      "               meta_A549 meta_foreskin_fibroblast   263.4437    1.0  -969.1395 1496.0269  False\n",
      "               meta_A549         meta_lymphoblast  -287.6376    1.0 -1520.2209  944.9456  False\n",
      "             meta_BEAS2B   meta_CD34_erythroblast    29.7489    1.0 -1202.8343 1262.3321  False\n",
      "             meta_BEAS2B          meta_CD4_T_cell     -6.102    1.0 -1238.6852 1226.4812  False\n",
      "             meta_BEAS2B                 meta_ESC   129.8791    1.0 -1102.7041 1362.4623  False\n",
      "             meta_BEAS2B                meta_G401  -114.9763    1.0 -1347.5596 1117.6069  False\n",
      "             meta_BEAS2B                meta_HAP1    164.468    1.0 -1068.1152 1397.0512  False\n",
      "             meta_BEAS2B              meta_HCT116    25.1406    1.0 -1207.4427 1257.7238  False\n",
      "             meta_BEAS2B      meta_HEK239T_HEK239   433.1281    1.0  -799.4551 1665.7113  False\n",
      "             meta_BEAS2B                meta_HeLa    -59.797    1.0 -1292.3802 1172.7862  False\n",
      "             meta_BEAS2B       meta_Jurkat_T_cell  1063.2455 0.2209  -169.3377 2295.8287  False\n",
      "             meta_BEAS2B                meta_K562  -199.3564    1.0 -1431.9396 1033.2268  False\n",
      "             meta_BEAS2B                meta_KBM7    17.4343    1.0 -1215.1489 1250.0175  False\n",
      "             meta_BEAS2B             meta_Kasumi1  -200.6297    1.0 -1433.2129 1031.9535  False\n",
      "             meta_BEAS2B               meta_LC2ad    46.0818    1.0 -1186.5014  1278.665  False\n",
      "             meta_BEAS2B                meta_MCF7    17.5447    1.0 -1215.0385 1250.1279  False\n",
      "             meta_BEAS2B              meta_MEL624    34.0791    1.0 -1198.5041 1266.6623  False\n",
      "             meta_BEAS2B              meta_NUDUL1   111.5271    1.0 -1121.0561 1344.1103  False\n",
      "             meta_BEAS2B              meta_OCILY1   874.3575 0.6374  -358.2258 2106.9407  False\n",
      "             meta_BEAS2B               meta_Ramos  -198.5361    1.0 -1431.1194 1034.0471  False\n",
      "             meta_BEAS2B              meta_S2VP10     5.4648    1.0 -1227.1184  1238.048  False\n",
      "             meta_BEAS2B              meta_SUDHL4  -433.9053    1.0 -1666.4886  798.6779  False\n",
      "             meta_BEAS2B                meta_THP1  -355.5224    1.0 -1588.1056  877.0608  False\n",
      "             meta_BEAS2B                meta_U936  -703.5353 0.9387 -1936.1185  529.0479  False\n",
      "             meta_BEAS2B meta_foreskin_fibroblast   247.9597    1.0  -984.6235 1480.5429  False\n",
      "             meta_BEAS2B         meta_lymphoblast  -303.1217    1.0 -1535.7049  929.4615  False\n",
      "  meta_CD34_erythroblast          meta_CD4_T_cell   -35.8509    1.0 -1268.4341 1196.7323  False\n",
      "  meta_CD34_erythroblast                 meta_ESC   100.1302    1.0  -1132.453 1332.7134  False\n",
      "  meta_CD34_erythroblast                meta_G401  -144.7252    1.0 -1377.3084  1087.858  False\n",
      "  meta_CD34_erythroblast                meta_HAP1   134.7191    1.0 -1097.8641 1367.3023  False\n",
      "  meta_CD34_erythroblast              meta_HCT116    -4.6083    1.0 -1237.1915 1227.9749  False\n",
      "  meta_CD34_erythroblast      meta_HEK239T_HEK239   403.3792    1.0   -829.204 1635.9624  False\n",
      "  meta_CD34_erythroblast                meta_HeLa   -89.5459    1.0 -1322.1291 1143.0373  False\n",
      "  meta_CD34_erythroblast       meta_Jurkat_T_cell  1033.4966 0.2729  -199.0866 2266.0799  False\n",
      "  meta_CD34_erythroblast                meta_K562  -229.1053    1.0 -1461.6885 1003.4779  False\n",
      "  meta_CD34_erythroblast                meta_KBM7   -12.3146    1.0 -1244.8978 1220.2686  False\n",
      "  meta_CD34_erythroblast             meta_Kasumi1  -230.3786    1.0 -1462.9618 1002.2046  False\n",
      "  meta_CD34_erythroblast               meta_LC2ad    16.3329    1.0 -1216.2503 1248.9161  False\n",
      "  meta_CD34_erythroblast                meta_MCF7   -12.2042    1.0 -1244.7874 1220.3791  False\n",
      "  meta_CD34_erythroblast              meta_MEL624     4.3302    1.0  -1228.253 1236.9134  False\n",
      "  meta_CD34_erythroblast              meta_NUDUL1    81.7782    1.0  -1150.805 1314.3614  False\n",
      "  meta_CD34_erythroblast              meta_OCILY1   844.6086 0.7072  -387.9746 2077.1918  False\n",
      "  meta_CD34_erythroblast               meta_Ramos   -228.285    1.0 -1460.8682 1004.2982  False\n",
      "  meta_CD34_erythroblast              meta_S2VP10   -24.2841    1.0 -1256.8673 1208.2991  False\n",
      "  meta_CD34_erythroblast              meta_SUDHL4  -463.6542 0.9999 -1696.2374   768.929  False\n",
      "  meta_CD34_erythroblast                meta_THP1  -385.2713    1.0 -1617.8545  847.3119  False\n",
      "  meta_CD34_erythroblast                meta_U936  -733.2842 0.9068 -1965.8674  499.2991  False\n",
      "  meta_CD34_erythroblast meta_foreskin_fibroblast   218.2108    1.0 -1014.3724  1450.794  False\n",
      "  meta_CD34_erythroblast         meta_lymphoblast  -332.8706    1.0 -1565.4538  899.7127  False\n",
      "         meta_CD4_T_cell                 meta_ESC   135.9811    1.0 -1096.6022 1368.5643  False\n",
      "         meta_CD4_T_cell                meta_G401  -108.8743    1.0 -1341.4576 1123.7089  False\n",
      "         meta_CD4_T_cell                meta_HAP1     170.57    1.0 -1062.0132 1403.1532  False\n",
      "         meta_CD4_T_cell              meta_HCT116    31.2425    1.0 -1201.3407 1263.8258  False\n",
      "         meta_CD4_T_cell      meta_HEK239T_HEK239   439.2301 0.9999  -793.3532 1671.8133  False\n",
      "         meta_CD4_T_cell                meta_HeLa    -53.695    1.0 -1286.2782 1178.8882  False\n",
      "         meta_CD4_T_cell       meta_Jurkat_T_cell  1069.3475 0.2111  -163.2357 2301.9307  False\n",
      "         meta_CD4_T_cell                meta_K562  -193.2544    1.0 -1425.8376 1039.3288  False\n",
      "         meta_CD4_T_cell                meta_KBM7    23.5363    1.0 -1209.0469 1256.1195  False\n",
      "         meta_CD4_T_cell             meta_Kasumi1  -194.5277    1.0 -1427.1109 1038.0555  False\n",
      "         meta_CD4_T_cell               meta_LC2ad    52.1838    1.0 -1180.3994  1284.767  False\n",
      "         meta_CD4_T_cell                meta_MCF7    23.6467    1.0 -1208.9365 1256.2299  False\n",
      "         meta_CD4_T_cell              meta_MEL624    40.1811    1.0 -1192.4022 1272.7643  False\n",
      "         meta_CD4_T_cell              meta_NUDUL1   117.6291    1.0 -1114.9541 1350.2123  False\n",
      "         meta_CD4_T_cell              meta_OCILY1   880.4594 0.6226  -352.1238 2113.0427  False\n",
      "         meta_CD4_T_cell               meta_Ramos  -192.4342    1.0 -1425.0174 1040.1491  False\n",
      "         meta_CD4_T_cell              meta_S2VP10    11.5668    1.0 -1221.0164   1244.15  False\n",
      "         meta_CD4_T_cell              meta_SUDHL4  -427.8034    1.0 -1660.3866  804.7799  False\n",
      "         meta_CD4_T_cell                meta_THP1  -349.4204    1.0 -1582.0037  883.1628  False\n",
      "         meta_CD4_T_cell                meta_U936  -697.4333 0.9441 -1930.0165  535.1499  False\n",
      "         meta_CD4_T_cell meta_foreskin_fibroblast   254.0617    1.0  -978.5215 1486.6449  False\n",
      "         meta_CD4_T_cell         meta_lymphoblast  -297.0197    1.0 -1529.6029  935.5635  False\n",
      "                meta_ESC                meta_G401  -244.8554    1.0 -1477.4386  987.7278  False\n",
      "                meta_ESC                meta_HAP1    34.5889    1.0 -1197.9943 1267.1722  False\n",
      "                meta_ESC              meta_HCT116  -104.7385    1.0 -1337.3217 1127.8447  False\n",
      "                meta_ESC      meta_HEK239T_HEK239    303.249    1.0  -929.3342 1535.8322  False\n",
      "                meta_ESC                meta_HeLa  -189.6761    1.0 -1422.2593 1042.9071  False\n",
      "                meta_ESC       meta_Jurkat_T_cell   933.3665 0.4928  -299.2168 2165.9497  False\n",
      "                meta_ESC                meta_K562  -329.2355    1.0 -1561.8187  903.3478  False\n",
      "                meta_ESC                meta_KBM7  -112.4448    1.0  -1345.028 1120.1384  False\n",
      "                meta_ESC             meta_Kasumi1  -330.5088    1.0  -1563.092  902.0745  False\n",
      "                meta_ESC               meta_LC2ad   -83.7973    1.0 -1316.3805 1148.7859  False\n",
      "                meta_ESC                meta_MCF7  -112.3343    1.0 -1344.9176 1120.2489  False\n",
      "                meta_ESC              meta_MEL624      -95.8    1.0 -1328.3832 1136.7832  False\n",
      "                meta_ESC              meta_NUDUL1    -18.352    1.0 -1250.9352 1214.2312  False\n",
      "                meta_ESC              meta_OCILY1   744.4784 0.8924  -488.1048 1977.0616  False\n",
      "                meta_ESC               meta_Ramos  -328.4152    1.0 -1560.9984   904.168  False\n",
      "                meta_ESC              meta_S2VP10  -124.4143    1.0 -1356.9975 1108.1689  False\n",
      "                meta_ESC              meta_SUDHL4  -563.7844 0.9964 -1796.3676  668.7988  False\n",
      "                meta_ESC                meta_THP1  -485.4015 0.9997 -1717.9847  747.1817  False\n",
      "                meta_ESC                meta_U936  -833.4143 0.7323 -2065.9976  399.1689  False\n",
      "                meta_ESC meta_foreskin_fibroblast   118.0806    1.0 -1114.5026 1350.6638  False\n",
      "                meta_ESC         meta_lymphoblast  -433.0007    1.0  -1665.584  799.5825  False\n",
      "               meta_G401                meta_HAP1   279.4443    1.0  -953.1389 1512.0276  False\n",
      "               meta_G401              meta_HCT116   140.1169    1.0 -1092.4663 1372.7001  False\n",
      "               meta_G401      meta_HEK239T_HEK239   548.1044 0.9977  -684.4788 1780.6876  False\n",
      "               meta_G401                meta_HeLa    55.1793    1.0 -1177.4039 1287.7625  False\n",
      "               meta_G401       meta_Jurkat_T_cell  1178.2219 0.0846   -54.3614 2410.8051  False\n",
      "               meta_G401                meta_K562   -84.3801    1.0 -1316.9633 1148.2032  False\n",
      "               meta_G401                meta_KBM7   132.4106    1.0 -1100.1726 1364.9938  False\n",
      "               meta_G401             meta_Kasumi1   -85.6534    1.0 -1318.2366 1146.9299  False\n",
      "               meta_G401               meta_LC2ad   161.0581    1.0 -1071.5251 1393.6414  False\n",
      "               meta_G401                meta_MCF7   132.5211    1.0 -1100.0622 1365.1043  False\n",
      "               meta_G401              meta_MEL624   149.0554    1.0 -1083.5278 1381.6386  False\n",
      "               meta_G401              meta_NUDUL1   226.5034    1.0 -1006.0798 1459.0866  False\n",
      "               meta_G401              meta_OCILY1   989.3338 0.3626  -243.2494  2221.917  False\n",
      "               meta_G401               meta_Ramos   -83.5598    1.0  -1316.143 1149.0234  False\n",
      "               meta_G401              meta_S2VP10   120.4411    1.0 -1112.1421 1353.0243  False\n",
      "               meta_G401              meta_SUDHL4   -318.929    1.0 -1551.5122  913.6542  False\n",
      "               meta_G401                meta_THP1  -240.5461    1.0 -1473.1293  992.0371  False\n",
      "               meta_G401                meta_U936  -588.5589 0.9933 -1821.1422  644.0243  False\n",
      "               meta_G401 meta_foreskin_fibroblast    362.936    1.0  -869.6472 1595.5192  False\n",
      "               meta_G401         meta_lymphoblast  -188.1453    1.0 -1420.7286 1044.4379  False\n",
      "               meta_HAP1              meta_HCT116  -139.3275    1.0 -1371.9107 1093.2558  False\n",
      "               meta_HAP1      meta_HEK239T_HEK239   268.6601    1.0  -963.9232 1501.2433  False\n",
      "               meta_HAP1                meta_HeLa   -224.265    1.0 -1456.8482 1008.3182  False\n",
      "               meta_HAP1       meta_Jurkat_T_cell   898.7775 0.5778  -333.8057 2131.3607  False\n",
      "               meta_HAP1                meta_K562  -363.8244    1.0 -1596.4076  868.7588  False\n",
      "               meta_HAP1                meta_KBM7  -147.0337    1.0 -1379.6169 1085.5495  False\n",
      "               meta_HAP1             meta_Kasumi1  -365.0977    1.0 -1597.6809  867.4855  False\n",
      "               meta_HAP1               meta_LC2ad  -118.3862    1.0 -1350.9694  1114.197  False\n",
      "               meta_HAP1                meta_MCF7  -146.9233    1.0 -1379.5065 1085.6599  False\n",
      "               meta_HAP1              meta_MEL624  -130.3889    1.0 -1362.9722 1102.1943  False\n",
      "               meta_HAP1              meta_NUDUL1   -52.9409    1.0 -1285.5241 1179.6423  False\n",
      "               meta_HAP1              meta_OCILY1   709.8894 0.9326  -522.6938 1942.4727  False\n",
      "               meta_HAP1               meta_Ramos  -363.0042    1.0 -1595.5874  869.5791  False\n",
      "               meta_HAP1              meta_S2VP10  -159.0032    1.0 -1391.5864   1073.58  False\n",
      "               meta_HAP1              meta_SUDHL4  -598.3734 0.9915 -1830.9566  634.2099  False\n",
      "               meta_HAP1                meta_THP1  -519.9904  0.999 -1752.5737  712.5928  False\n",
      "               meta_HAP1                meta_U936  -868.0033 0.6526 -2100.5865  364.5799  False\n",
      "               meta_HAP1 meta_foreskin_fibroblast    83.4917    1.0 -1149.0915 1316.0749  False\n",
      "               meta_HAP1         meta_lymphoblast  -467.5897 0.9998 -1700.1729  764.9935  False\n",
      "             meta_HCT116      meta_HEK239T_HEK239   407.9875    1.0  -824.5957 1640.5707  False\n",
      "             meta_HCT116                meta_HeLa   -84.9376    1.0 -1317.5208 1147.6456  False\n",
      "             meta_HCT116       meta_Jurkat_T_cell   1038.105 0.2644  -194.4782 2270.6882  False\n",
      "             meta_HCT116                meta_K562   -224.497    1.0 -1457.0802 1008.0863  False\n",
      "             meta_HCT116                meta_KBM7    -7.7063    1.0 -1240.2895 1224.8769  False\n",
      "             meta_HCT116             meta_Kasumi1  -225.7702    1.0 -1458.3535  1006.813  False\n",
      "             meta_HCT116               meta_LC2ad    20.9412    1.0  -1211.642 1253.5245  False\n",
      "             meta_HCT116                meta_MCF7    -7.5958    1.0 -1240.1791 1224.9874  False\n",
      "             meta_HCT116              meta_MEL624     8.9385    1.0 -1223.6447 1241.5217  False\n",
      "             meta_HCT116              meta_NUDUL1    86.3865    1.0 -1146.1967 1318.9697  False\n",
      "             meta_HCT116              meta_OCILY1   849.2169 0.6967  -383.3663 2081.8001  False\n",
      "             meta_HCT116               meta_Ramos  -223.6767    1.0 -1456.2599 1008.9065  False\n",
      "             meta_HCT116              meta_S2VP10   -19.6758    1.0  -1252.259 1212.9074  False\n",
      "             meta_HCT116              meta_SUDHL4  -459.0459 0.9999 -1691.6291  773.5373  False\n",
      "             meta_HCT116                meta_THP1   -380.663    1.0 -1613.2462  851.9202  False\n",
      "             meta_HCT116                meta_U936  -728.6758 0.9124 -1961.2591  503.9074  False\n",
      "             meta_HCT116 meta_foreskin_fibroblast   222.8191    1.0 -1009.7641 1455.4023  False\n",
      "             meta_HCT116         meta_lymphoblast  -328.2622    1.0 -1560.8455   904.321  False\n",
      "     meta_HEK239T_HEK239                meta_HeLa  -492.9251 0.9996 -1725.5083  739.6581  False\n",
      "     meta_HEK239T_HEK239       meta_Jurkat_T_cell   630.1175 0.9831  -602.4658 1862.7007  False\n",
      "     meta_HEK239T_HEK239                meta_K562  -632.4845 0.9823 -1865.0677  600.0987  False\n",
      "     meta_HEK239T_HEK239                meta_KBM7  -415.6938    1.0  -1648.277  816.8894  False\n",
      "     meta_HEK239T_HEK239             meta_Kasumi1  -633.7578 0.9818  -1866.341  598.8255  False\n",
      "     meta_HEK239T_HEK239               meta_LC2ad  -387.0463    1.0 -1619.6295  845.5369  False\n",
      "     meta_HEK239T_HEK239                meta_MCF7  -415.5834    1.0 -1648.1666  816.9999  False\n",
      "     meta_HEK239T_HEK239              meta_MEL624   -399.049    1.0 -1631.6322  833.5342  False\n",
      "     meta_HEK239T_HEK239              meta_NUDUL1   -321.601    1.0 -1554.1842  910.9822  False\n",
      "     meta_HEK239T_HEK239              meta_OCILY1   441.2294 0.9999  -791.3538 1673.8126  False\n",
      "     meta_HEK239T_HEK239               meta_Ramos  -631.6642 0.9826 -1864.2474   600.919  False\n",
      "     meta_HEK239T_HEK239              meta_S2VP10  -427.6633    1.0 -1660.2465  804.9199  False\n",
      "     meta_HEK239T_HEK239              meta_SUDHL4  -867.0334 0.6549 -2099.6166  365.5498  False\n",
      "     meta_HEK239T_HEK239                meta_THP1  -788.6505 0.8225 -2021.2337  443.9327  False\n",
      "     meta_HEK239T_HEK239                meta_U936 -1136.6634 0.1227 -2369.2466   95.9199  False\n",
      "     meta_HEK239T_HEK239 meta_foreskin_fibroblast  -185.1684    1.0 -1417.7516 1047.4148  False\n",
      "     meta_HEK239T_HEK239         meta_lymphoblast  -736.2498 0.9031  -1968.833  496.3335  False\n",
      "               meta_HeLa       meta_Jurkat_T_cell  1123.0425 0.1378  -109.5407 2355.6258  False\n",
      "               meta_HeLa                meta_K562  -139.5594    1.0 -1372.1426 1093.0238  False\n",
      "               meta_HeLa                meta_KBM7    77.2313    1.0 -1155.3519 1309.8145  False\n",
      "               meta_HeLa             meta_Kasumi1  -140.8327    1.0 -1373.4159 1091.7505  False\n",
      "               meta_HeLa               meta_LC2ad   105.8788    1.0 -1126.7044  1338.462  False\n",
      "               meta_HeLa                meta_MCF7    77.3417    1.0 -1155.2415  1309.925  False\n",
      "               meta_HeLa              meta_MEL624    93.8761    1.0 -1138.7071 1326.4593  False\n",
      "               meta_HeLa              meta_NUDUL1   171.3241    1.0 -1061.2591 1403.9073  False\n",
      "               meta_HeLa              meta_OCILY1   934.1545 0.4909  -298.4287 2166.7377  False\n",
      "               meta_HeLa               meta_Ramos  -138.7391    1.0 -1371.3223 1093.8441  False\n",
      "               meta_HeLa              meta_S2VP10    65.2618    1.0 -1167.3214  1297.845  False\n",
      "               meta_HeLa              meta_SUDHL4  -374.1083    1.0 -1606.6915  858.4749  False\n",
      "               meta_HeLa                meta_THP1  -295.7254    1.0 -1528.3086  936.8578  False\n",
      "               meta_HeLa                meta_U936  -643.7383 0.9779 -1876.3215   588.845  False\n",
      "               meta_HeLa meta_foreskin_fibroblast   307.7567    1.0  -924.8265 1540.3399  False\n",
      "               meta_HeLa         meta_lymphoblast  -243.3247    1.0 -1475.9079  989.2586  False\n",
      "      meta_Jurkat_T_cell                meta_K562 -1262.6019 0.0367 -2495.1851  -30.0187   True\n",
      "      meta_Jurkat_T_cell                meta_KBM7 -1045.8112 0.2505 -2278.3945   186.772  False\n",
      "      meta_Jurkat_T_cell             meta_Kasumi1 -1263.8752 0.0362 -2496.4584   -31.292   True\n",
      "      meta_Jurkat_T_cell               meta_LC2ad -1017.1637 0.3044 -2249.7469  215.4195  False\n",
      "      meta_Jurkat_T_cell                meta_MCF7 -1045.7008 0.2507  -2278.284  186.8824  False\n",
      "      meta_Jurkat_T_cell              meta_MEL624 -1029.1665 0.2811 -2261.7497  203.4168  False\n",
      "      meta_Jurkat_T_cell              meta_NUDUL1  -951.7184 0.4486 -2184.3017  280.8648  False\n",
      "      meta_Jurkat_T_cell              meta_OCILY1  -188.8881    1.0 -1421.4713 1043.6951  False\n",
      "      meta_Jurkat_T_cell               meta_Ramos -1261.7817  0.037 -2494.3649  -29.1985   True\n",
      "      meta_Jurkat_T_cell              meta_S2VP10 -1057.7807 0.2299  -2290.364  174.8025  False\n",
      "      meta_Jurkat_T_cell              meta_SUDHL4 -1497.1509 0.0022 -2729.7341 -264.5677   True\n",
      "      meta_Jurkat_T_cell                meta_THP1  -1418.768 0.0061 -2651.3512 -186.1847   True\n",
      "      meta_Jurkat_T_cell                meta_U936 -1766.7808    0.0  -2999.364 -534.1976   True\n",
      "      meta_Jurkat_T_cell meta_foreskin_fibroblast  -815.2858 0.7709 -2047.8691  417.2974  False\n",
      "      meta_Jurkat_T_cell         meta_lymphoblast -1366.3672 0.0115 -2598.9504  -133.784   True\n",
      "               meta_K562                meta_KBM7   216.7907    1.0 -1015.7925 1449.3739  False\n",
      "               meta_K562             meta_Kasumi1    -1.2733    1.0 -1233.8565 1231.3099  False\n",
      "               meta_K562               meta_LC2ad   245.4382    1.0   -987.145 1478.0214  False\n",
      "               meta_K562                meta_MCF7   216.9011    1.0 -1015.6821 1449.4843  False\n",
      "               meta_K562              meta_MEL624   233.4355    1.0  -999.1478 1466.0187  False\n",
      "               meta_K562              meta_NUDUL1   310.8835    1.0  -921.6997 1543.4667  False\n",
      "               meta_K562              meta_OCILY1  1073.7139 0.2043  -158.8694 2306.2971  False\n",
      "               meta_K562               meta_Ramos     0.8202    1.0  -1231.763 1233.4035  False\n",
      "               meta_K562              meta_S2VP10   204.8212    1.0  -1027.762 1437.4044  False\n",
      "               meta_K562              meta_SUDHL4  -234.5489    1.0 -1467.1322  998.0343  False\n",
      "               meta_K562                meta_THP1   -156.166    1.0 -1388.7492 1076.4172  False\n",
      "               meta_K562                meta_U936  -504.1789 0.9994 -1736.7621  728.4043  False\n",
      "               meta_K562 meta_foreskin_fibroblast   447.3161 0.9999  -785.2671 1679.8993  False\n",
      "               meta_K562         meta_lymphoblast  -103.7653    1.0 -1336.3485 1128.8179  False\n",
      "               meta_KBM7             meta_Kasumi1   -218.064    1.0 -1450.6472 1014.5192  False\n",
      "               meta_KBM7               meta_LC2ad    28.6475    1.0 -1203.9357 1261.2307  False\n",
      "               meta_KBM7                meta_MCF7     0.1104    1.0 -1232.4728 1232.6937  False\n",
      "               meta_KBM7              meta_MEL624    16.6448    1.0 -1215.9384  1249.228  False\n",
      "               meta_KBM7              meta_NUDUL1    94.0928    1.0 -1138.4904  1326.676  False\n",
      "               meta_KBM7              meta_OCILY1   856.9232 0.6788    -375.66 2089.5064  False\n",
      "               meta_KBM7               meta_Ramos  -215.9704    1.0 -1448.5537 1016.6128  False\n",
      "               meta_KBM7              meta_S2VP10   -11.9695    1.0 -1244.5527 1220.6137  False\n",
      "               meta_KBM7              meta_SUDHL4  -451.3396 0.9999 -1683.9228  781.2436  False\n",
      "               meta_KBM7                meta_THP1  -372.9567    1.0 -1605.5399  859.6265  False\n",
      "               meta_KBM7                meta_U936  -720.9696 0.9211 -1953.5528  511.6137  False\n",
      "               meta_KBM7 meta_foreskin_fibroblast   230.5254    1.0 -1002.0578 1463.1086  False\n",
      "               meta_KBM7         meta_lymphoblast   -320.556    1.0 -1553.1392  912.0273  False\n",
      "            meta_Kasumi1               meta_LC2ad   246.7115    1.0  -985.8717 1479.2947  False\n",
      "            meta_Kasumi1                meta_MCF7   218.1744    1.0 -1014.4088 1450.7576  False\n",
      "            meta_Kasumi1              meta_MEL624   234.7088    1.0  -997.8745  1467.292  False\n",
      "            meta_Kasumi1              meta_NUDUL1   312.1568    1.0  -920.4264   1544.74  False\n",
      "            meta_Kasumi1              meta_OCILY1  1074.9871 0.2023  -157.5961 2307.5704  False\n",
      "            meta_Kasumi1               meta_Ramos     2.0935    1.0 -1230.4897 1234.6768  False\n",
      "            meta_Kasumi1              meta_S2VP10   206.0945    1.0 -1026.4887 1438.6777  False\n",
      "            meta_Kasumi1              meta_SUDHL4  -233.2757    1.0 -1465.8589  999.3076  False\n",
      "            meta_Kasumi1                meta_THP1  -154.8927    1.0 -1387.4759 1077.6905  False\n",
      "            meta_Kasumi1                meta_U936  -502.9056 0.9994 -1735.4888  729.6776  False\n",
      "            meta_Kasumi1 meta_foreskin_fibroblast   448.5894 0.9999  -783.9938 1681.1726  False\n",
      "            meta_Kasumi1         meta_lymphoblast   -102.492    1.0 -1335.0752 1130.0912  False\n",
      "              meta_LC2ad                meta_MCF7   -28.5371    1.0 -1261.1203 1204.0461  False\n",
      "              meta_LC2ad              meta_MEL624   -12.0027    1.0 -1244.5859 1220.5805  False\n",
      "              meta_LC2ad              meta_NUDUL1    65.4453    1.0 -1167.1379 1298.0285  False\n",
      "              meta_LC2ad              meta_OCILY1   828.2757 0.7435  -404.3076 2060.8589  False\n",
      "              meta_LC2ad               meta_Ramos   -244.618    1.0 -1477.2012  987.9653  False\n",
      "              meta_LC2ad              meta_S2VP10    -40.617    1.0 -1273.2002 1191.9662  False\n",
      "              meta_LC2ad              meta_SUDHL4  -479.9871 0.9997 -1712.5704  752.5961  False\n",
      "              meta_LC2ad                meta_THP1  -401.6042    1.0 -1634.1874   830.979  False\n",
      "              meta_LC2ad                meta_U936  -749.6171 0.8854 -1982.2003  482.9661  False\n",
      "              meta_LC2ad meta_foreskin_fibroblast   201.8779    1.0 -1030.7053 1434.4611  False\n",
      "              meta_LC2ad         meta_lymphoblast  -349.2035    1.0 -1581.7867  883.3797  False\n",
      "               meta_MCF7              meta_MEL624    16.5343    1.0 -1216.0489 1249.1176  False\n",
      "               meta_MCF7              meta_NUDUL1    93.9824    1.0 -1138.6009 1326.5656  False\n",
      "               meta_MCF7              meta_OCILY1   856.8127 0.6791  -375.7705 2089.3959  False\n",
      "               meta_MCF7               meta_Ramos  -216.0809    1.0 -1448.6641 1016.5023  False\n",
      "               meta_MCF7              meta_S2VP10   -12.0799    1.0 -1244.6632 1220.5033  False\n",
      "               meta_MCF7              meta_SUDHL4  -451.4501 0.9999 -1684.0333  781.1332  False\n",
      "               meta_MCF7                meta_THP1  -373.0671    1.0 -1605.6504  859.5161  False\n",
      "               meta_MCF7                meta_U936    -721.08  0.921 -1953.6632  511.5032  False\n",
      "               meta_MCF7 meta_foreskin_fibroblast    230.415    1.0 -1002.1682 1462.9982  False\n",
      "               meta_MCF7         meta_lymphoblast  -320.6664    1.0 -1553.2496  911.9168  False\n",
      "             meta_MEL624              meta_NUDUL1     77.448    1.0 -1155.1352 1310.0312  False\n",
      "             meta_MEL624              meta_OCILY1   840.2784  0.717  -392.3048 2072.8616  False\n",
      "             meta_MEL624               meta_Ramos  -232.6152    1.0 -1465.1984   999.968  False\n",
      "             meta_MEL624              meta_S2VP10   -28.6143    1.0 -1261.1975 1203.9689  False\n",
      "             meta_MEL624              meta_SUDHL4  -467.9844 0.9998 -1700.5676  764.5988  False\n",
      "             meta_MEL624                meta_THP1  -389.6015    1.0 -1622.1847  842.9817  False\n",
      "             meta_MEL624                meta_U936  -737.6143 0.9014 -1970.1976  494.9689  False\n",
      "             meta_MEL624 meta_foreskin_fibroblast   213.8806    1.0 -1018.7026 1446.4638  False\n",
      "             meta_MEL624         meta_lymphoblast  -337.2007    1.0  -1569.784  895.3825  False\n",
      "             meta_NUDUL1              meta_OCILY1   762.8304 0.8659  -469.7528 1995.4136  False\n",
      "             meta_NUDUL1               meta_Ramos  -310.0632    1.0 -1542.6464    922.52  False\n",
      "             meta_NUDUL1              meta_S2VP10  -106.0623    1.0 -1338.6455 1126.5209  False\n",
      "             meta_NUDUL1              meta_SUDHL4  -545.4324 0.9978 -1778.0156  687.1508  False\n",
      "             meta_NUDUL1                meta_THP1  -467.0495 0.9998 -1699.6327  765.5337  False\n",
      "             meta_NUDUL1                meta_U936  -815.0624 0.7713 -2047.6456  417.5209  False\n",
      "             meta_NUDUL1 meta_foreskin_fibroblast   136.4326    1.0 -1096.1506 1369.0158  False\n",
      "             meta_NUDUL1         meta_lymphoblast  -414.6488    1.0  -1647.232  817.9345  False\n",
      "             meta_OCILY1               meta_Ramos -1072.8936 0.2055 -2305.4768  159.6896  False\n",
      "             meta_OCILY1              meta_S2VP10  -868.8927 0.6505 -2101.4759  363.6905  False\n",
      "             meta_OCILY1              meta_SUDHL4 -1308.2628 0.0224  -2540.846  -75.6796   True\n",
      "             meta_OCILY1                meta_THP1 -1229.8799 0.0514 -2462.4631    2.7033  False\n",
      "             meta_OCILY1                meta_U936 -1577.8927 0.0007 -2810.4759 -345.3095   True\n",
      "             meta_OCILY1 meta_foreskin_fibroblast  -626.3978 0.9844  -1858.981  606.1855  False\n",
      "             meta_OCILY1         meta_lymphoblast -1177.4791 0.0852 -2410.0623   55.1041  False\n",
      "              meta_Ramos              meta_S2VP10   204.0009    1.0 -1028.5823 1436.5841  False\n",
      "              meta_Ramos              meta_SUDHL4  -235.3692    1.0 -1467.9524   997.214  False\n",
      "              meta_Ramos                meta_THP1  -156.9863    1.0 -1389.5695 1075.5969  False\n",
      "              meta_Ramos                meta_U936  -504.9991 0.9994 -1737.5823  727.5841  False\n",
      "              meta_Ramos meta_foreskin_fibroblast   446.4958 0.9999  -786.0874 1679.0791  False\n",
      "              meta_Ramos         meta_lymphoblast  -104.5855    1.0 -1337.1687 1127.9977  False\n",
      "             meta_S2VP10              meta_SUDHL4  -439.3701 0.9999 -1671.9533  793.2131  False\n",
      "             meta_S2VP10                meta_THP1  -360.9872    1.0 -1593.5704   871.596  False\n",
      "             meta_S2VP10                meta_U936  -709.0001 0.9335 -1941.5833  523.5832  False\n",
      "             meta_S2VP10 meta_foreskin_fibroblast   242.4949    1.0  -990.0883 1475.0781  False\n",
      "             meta_S2VP10         meta_lymphoblast  -308.5865    1.0 -1541.1697  923.9968  False\n",
      "             meta_SUDHL4                meta_THP1    78.3829    1.0 -1154.2003 1310.9661  False\n",
      "             meta_SUDHL4                meta_U936  -269.6299    1.0 -1502.2132  962.9533  False\n",
      "             meta_SUDHL4 meta_foreskin_fibroblast    681.865 0.9564  -550.7182 1914.4482  False\n",
      "             meta_SUDHL4         meta_lymphoblast   130.7837    1.0 -1101.7996 1363.3669  False\n",
      "               meta_THP1                meta_U936  -348.0129    1.0 -1580.5961  884.5704  False\n",
      "               meta_THP1 meta_foreskin_fibroblast   603.4821 0.9905  -629.1011 1836.0653  False\n",
      "               meta_THP1         meta_lymphoblast    52.4007    1.0 -1180.1825  1284.984  False\n",
      "               meta_U936 meta_foreskin_fibroblast    951.495 0.4491  -281.0882 2184.0782  False\n",
      "               meta_U936         meta_lymphoblast   400.4136    1.0  -832.1696 1632.9968  False\n",
      "meta_foreskin_fibroblast         meta_lymphoblast  -551.0814 0.9975 -1783.6646  681.5019  False\n",
      "-----------------------------------------------------------------------------------------------\n"
     ]
    }
   ],
   "source": [
    "fit_parses = [\n",
    "    meta_OCILY1_fp,\n",
    "    meta_G401_fp,\n",
    "    meta_HCT116_fp,\n",
    "    meta_MEL624_fp,\n",
    "    meta_HeLa_fp,\n",
    "    meta_THP1_fp,\n",
    "    meta_foreskin_fibroblast_fp,\n",
    "    meta_A375_fp,\n",
    "    meta_SUDHL4_fp,\n",
    "    meta_lymphoblast_fp,\n",
    "    meta_Jurkat_T_cell_fp,\n",
    "    meta_ESC_fp,\n",
    "    meta_LC2ad_fp,\n",
    "    meta_Ramos_fp,\n",
    "    meta_HEK239T_HEK239_fp,\n",
    "    meta_BEAS2B_fp,\n",
    "    meta_NUDUL1_fp,\n",
    "    meta_K562_fp,\n",
    "    meta_MCF7_fp,\n",
    "    meta_Kasumi1_fp,\n",
    "    meta_S2VP10_fp,\n",
    "    meta_KBM7_fp,\n",
    "    meta_CD34_erythroblast_fp,\n",
    "    meta_HAP1_fp,\n",
    "    meta_U936_fp,\n",
    "    meta_CD4_T_cell_fp,\n",
    "    meta_A549_fp\n",
    "]\n",
    "\n",
    "treatments = [\n",
    "    'meta_OCILY1',\n",
    "    'meta_G401',\n",
    "    'meta_HCT116',\n",
    "    'meta_MEL624',\n",
    "    'meta_HeLa',\n",
    "    'meta_THP1',\n",
    "    'meta_foreskin_fibroblast',\n",
    "    'meta_A375',\n",
    "    'meta_SUDHL4',\n",
    "    'meta_lymphoblast',\n",
    "    'meta_Jurkat_T_cell',\n",
    "    'meta_ESC',\n",
    "    'meta_LC2ad',\n",
    "    'meta_Ramos',\n",
    "    'meta_HEK239T_HEK239',\n",
    "    'meta_BEAS2B',\n",
    "    'meta_NUDUL1',\n",
    "    'meta_K562',\n",
    "    'meta_MCF7',\n",
    "    'meta_Kasumi1',\n",
    "    'meta_S2VP10',\n",
    "    'meta_KBM7',\n",
    "    'meta_CD34_erythroblast',\n",
    "    'meta_HAP1',\n",
    "    'meta_U936',\n",
    "    'meta_CD4_T_cell',\n",
    "    'meta_A549'\n",
    "]\n",
    "\n",
    "print(len(treatments))\n",
    "\n",
    "liet_param = 'sT'\n",
    "\n",
    "anova_inputs = anova_data_formatter(liet_param, fit_parses, treatments)\n",
    "anova_inputs['subject'] = anova_inputs['subject'].astype('category')\n",
    "\n",
    "aovrm = AnovaRM(anova_inputs, liet_param, 'subject', within=['treatment'], aggregate_func='mean')\n",
    "\n",
    "res = aovrm.fit()\n",
    "print(res)\n",
    "\n",
    "# Perform Tukey's HSD post-hoc test\n",
    "tukey = pairwise_tukeyhsd(endog=anova_inputs[liet_param], groups=anova_inputs['treatment'], alpha=0.05)\n",
    "\n",
    "# Display the results\n",
    "print(tukey)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "statsmodels",
   "language": "python",
   "name": "statsmodels"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
